checkov/kubernetes/graph_builder/graph_components/ResourceKeywordIdentifier.py

```diff
@@ -1,9 +1,30 @@
 class ResourceKeywordIdentifier:
+    """
+    this class maps connections between resources by their unique keyword identifier.
+    each resource in this class has a list of objects, and each object defines a potential connection to a different
+    resource only if all attributes in the object are matched. each object in the list is independent regardless of
+    the other objects in the list.
+
+    for example:
+    A ServiceAccount resource A with the property 'metadata.name' with value 'service-123' will match a resource B
+    of type 'ClusterRoleBinding' with a 'subjects.name' property equals to 'service-123'
+    and
+    the property 'kind' for resource A matches resource B's 'subjects.kind' property
+    """
+
     KINDS_KEYWORDS_MAP = {
-        "PersistentVolumeClaim": ["claimName"],
-        "ServiceAccount": ["serviceAccountName"],
-        "ClusterRole": ["rules[].resources", "rules[].resourceNames"],
-        "ClusterRoleBinding": [["roleRef.name", "kind"], ["subjects[].name", "kind"], ["aggregationRule.clusterRoleSelectors.matchLabels"]],
-        "Role": ["rules[].resources", "rules[].resourceNames"],
-        "RoleBinding": [["roleRef.name", "kind"], ["subjects[].name", "kind"]]
+        # "PersistentVolumeClaim": ["claimName"],
+        # "ClusterRole": ["rules[].resources", "rules[].resourceNames"],
+        # "Role": ["rules[].resources", "rules[].resourceNames"],
```

Comment:
can this be removed or is it left for tackling later?

Reply:
yes, it's a TODO place holder for resources that will probably need to be handled later on